bin/compile

```diff
@@ -1,77 +1,105 @@
 #!/usr/bin/env bash
 
+set -euo pipefail
+
 function log() {
-        echo "-----> $*"
+    echo "-----> $*"
 }
 
-export BUILD_DIR=$1
-export ENV_DIR=$3
-export POETRY_VERSION="$(cat $ENV_DIR/POETRY_VERSION 2>/dev/null)"
+function indent() {
+    sed -e 's/^/       /'
+}
+
+export BUILD_DIR="$1"
+export ENV_DIR="$3"
+
+BUILDPACK_VARIABLES="DISABLE_POETRY_CREATE_RUNTIME_FILE PYTHON_RUNTIME_VERSION"
 
-export DISABLE_POETRY_CREATE_RUNTIME_FILE="$(cat $ENV_DIR/DISABLE_POETRY_CREATE_RUNTIME_FILE 2>/dev/null || true)"
+shopt -s nullglob
 
-if [ -z "$POETRY_VERSION" ];
-then
+for env_file in "$ENV_DIR"/POETRY_* ; do
+    export "$(basename "$env_file")=$(cat "$env_file" 2>/dev/null)"
+done
+
+for env_file in $BUILDPACK_VARIABLES ; do
+    if [ -f "$ENV_DIR/$env_file" ] ; then
+        export "$(basename "$env_file")=$(cat "$ENV_DIR/$env_file" 2>/dev/null)"
+    fi
+done
+
+if [ -z "${POETRY_VERSION:-}" ] ; then
```

Comment:
A common strategy for ensuring default values is to `VAR="${VAR:-default}"` at the top of scripts.

Reply:
That's correct, but here we want to warn explicitly that the variable has been defaulted.